go-api/utils/utils.go

```diff
@@ -0,0 +1,209 @@
+package utils
+
+import (
+	"context"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+
+	"golang.org/x/crypto/scrypt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/middleware/cors"
+	"github.com/gofiber/fiber/v2/middleware/logger"
+	"github.com/jackc/pgx/v5"
+	"github.com/redis/go-redis/v9"
+)
+
+func GetPgx(c *fiber.Ctx) pgx.Conn {
+	return c.Locals("pgxConn").(pgx.Conn)
+}
+
+func GetRdb(c *fiber.Ctx) redis.Conn {
+	return c.Locals("rdb").(redis.Conn)
+}
+
+var ctx = context.Background()
+
+func SetRedis(c *fiber.Ctx, key string, value string) error {
+	redisConn := GetRdb(c)
+
+	_err := redisConn.Set(c.Context(), key, value, 0).Err()
+	if _err != nil {
+		return _err
+	}
+	return nil
+}
+
+func GetRedis(c *fiber.Ctx, key string) (string, error) {
+	redisConn := GetRdb(c)
+	val, _err := redisConn.Get(c.Context(), key).Result()
+	if _err != nil {
+		return "", _err
+	}
+	return val, nil
+}
+
+func RawQueryWithoutReturn(c *fiber.Ctx, query string, args map[string]any) error {
+	pgxConn := GetPgx(c)
+	rows, _err := pgxConn.Query(ctx, query, pgx.NamedArgs(args))
+	if _err != nil {
+		return _err
+	}
+	rows.Close()
+
+	return nil
+}
+
+func QueryRow[T any](c *fiber.Ctx, query string, args map[string]any) (T, error) {
+	var result T
+	pgxConn := GetPgx(c)
+
+	rows, _err := pgxConn.Query(ctx, query, pgx.NamedArgs(args))
+	if _err != nil {
+		return result, _err
+	}
+
+	result, _err = pgx.CollectExactlyOneRow(rows, pgx.RowToStructByName[T])
+	if _err != nil {
+		return result, _err
+	}
+
+	return result, nil
+}
+
+func QueryRows[T any](c *fiber.Ctx, query string, args map[string]any) ([]T, error) {
+	results := []T{}
+	pgxConn := GetPgx(c)
+
+	rows, err := pgxConn.Query(ctx, query, pgx.NamedArgs(args))
+	if err != nil {
+		return results, err
+	}
+
+	results, err = pgx.CollectRows(rows, pgx.RowToStructByName[T])
+	if err != nil {
+		return results, err
+	}
+
+	return results, nil
+}
+
+func Setup() (*pgx.Conn, *redis.Conn, *fiber.App, error) {
+	config := LoadEnvVars()
+	// pgsql connect
+	pgxConn, pgxError := pgx.Connect(context.Background(), config["DATABASE_URL"].(string))
+	if pgxError != nil {
+		return nil, nil, nil, pgxError
+	}
+	// redis connect
+	rdb := redis.NewClient(&redis.Options{
+		Addr: config["REDIS_HOST"].(string) + ":" + config["REDIS_PORT"].(string),
+	}).Conn()
+	_, rdbErr := rdb.Ping(context.Background()).Result()
+	if rdbErr != nil {
+		return nil, nil, nil, rdbErr
+	}
+	app := fiber.New(fiber.Config{
+		AppName:           "go-api v0.0.1",
```

Comment:
I saw there is a `.version` file with different version number. We should have a single way to setup a version number. Have you thought about it?

Reply:
I didn't think about it, just added file since previous gitgub action deployment script took version from package.json file. thought it had to be written somewhere to be reference from github action. maybe I can make it reference from .version file here